### PR TITLE
Swift 4 migration

### PR DESCRIPTION
**Pull request by lucianomarisi**

### CODE REVIEW

File: Example/AppDelegate.swift [hunk 1]
```diff
@@ -21,7 +21,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
   private func setupNetworkActivityHandler(for application: UIApplication) {
     NetworkServiceActivity.activityChangeHandler = { isActive in
-      application.isNetworkActivityIndicatorVisible = isActive
+      DispatchQueue.main.async {
```
Comment (KaneCheshire) [inline Example/AppDelegate.swift]: Ooh that thread sanitiser catches so much. Love it.

**Review by KaneCheshire — APPROVED** [review]

File: Example/AppDelegate.swift (hunk 1 above)
Comment (KITSTABLucianoMarisi) [reply Example/AppDelegate.swift]: It's awesome! I was thinking that we probably should call this block on the main thread within the library, thoughts?

Comment (lucianomarisi) [reply Example/AppDelegate.swift]: It's awesome! I was thinking that we probably should call this block on the main thread within the library, thoughts?

Comment (KaneCheshire) [reply Example/AppDelegate.swift]: Would it be possible to let people choose what queue to dispatch on, defaulting to `main`?

Comment (lucianomarisi) [reply Example/AppDelegate.swift]: Potentially we could expose a static property on `NetworkServiceActivity` to set the queue, which is originally set to main. I'm wondering what would be the benefit of this though, as in under what circumstances would you not want this dispatched on the main thread?

Comment (KaneCheshire) [reply Example/AppDelegate.swift]: Yeah you're right actually. Stick with main!

**Review by Neil3079 — APPROVED** [review]

**Pull request merged**